File: ChatGPT/python/output/output_02b.py
# ChatGPT-generated solution will go here.
def is_safe(report):
    diffs = [b - a for a, b in zip(report, report[1:])]
    if all(d > 0 and 1 <= d <= 3 for d in diffs):
        return True
    if all(d < 0 and -3 <= d <= -1 for d in diffs):
        return True
    return False

def is_safe_with_dampener(report):
    for i in range(len(report)):
        reduced = report[:i] + report[i+1:]
        if is_safe(reduced):
            return True
    return False

def main():
    safe_count = 0
    with open('input.txt') as f:
        for line in f:
            if line.strip():
                levels = list(map(int, line.strip().split()))
                if is_safe(levels) or is_safe_with_dampener(levels):
                    safe_count += 1
    print(safe_count)

if __name__ == '__main__':
    main()

File: ChatGPT/clojure/output/output_02b.clj
;; ChatGPT-generated solution will go here.
(ns output_02b
  (:require [clojure.string :as str]))

(defn safe-report? [nums]
  (let [deltas (map - (rest nums) nums)
        signs (set (map compare deltas (repeat 0)))]
    (and (or (= signs #{1}) (= signs #{-1}))
         (every? #(<= 1 (Math/abs %) 3) deltas))))

(defn dampened-safe? [nums]
  (some safe-report? (map #(vec (concat (subvec nums 0 %) (subvec nums (inc %)))) (range (count nums)))))

(defn part [lines]
  (->> lines
       (map #(vec (map parse-long (str/split % #"\s+"))))
       (filter #(or (safe-report? %) (dampened-safe? %)))
       count))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
